File: 2463-minimum-recolors-to-get-k-consecutive-black-blocks/2463-minimum-recolors-to-get-k-consecutive-black-blocks.py
class Solution:
    def minimumRecolors(self, blocks: str, k: int) -> int:
        res = float('inf')
        n = blocks[:k].count('W')
        res = min(res, n)
        
        for i in range(k, len(blocks)):
            n -= (blocks[i - k] == 'W')
            n += (blocks[i] == 'W')
            res = min(res, n)
        
        return res

File: 2463-minimum-recolors-to-get-k-consecutive-black-blocks/2463-minimum-recolors-to-get-k-consecutive-black-blocks.java
class Solution {
    public int minimumRecolors(String blocks, int k) {
        int res = Integer.MAX_VALUE,n = 0;
        for(int i = 0;i < k;i++){
            if(blocks.charAt(i) == 'W'){
                n++;
            }
        }
        res = Math.min(res,n);
        for(int i = k;i < blocks.length();i++){
            n -= (blocks.charAt(i - k) == 'W')?1:0;
            n += (blocks.charAt(i) == 'W')?1:0;
            res = Math.min(res,n); 
        }
        return res;
    }
}